master/buildbot/db/builds.py

```diff
@@ -110,12 +116,13 @@ def thd(conn):
             while remaining:
                 batch, remaining = remaining[:100], remaining[100:]
                 q = tbl.update(whereclause=(tbl.c.id.in_(batch)))
-                conn.execute(q, finish_time=now)
-
+                with conn_execute(conn, q, finish_time=now):
+                    pass
```

Comment:
This looks very strange.

I think the problem is that you are running the context code in the `with` line (before entering the context manager), which is a bit unusual.

What I expected was something like:

```
with getconn() as c:
    c.execute(tbl.update(whereclause=(tbl.c.id.in_(batch))), finish_time=now)
```